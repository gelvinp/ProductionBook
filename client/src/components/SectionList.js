import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { Button, Icon, List } from 'semantic-ui-react'
import FileList from '../containers/FileListContainer.js'

//TODO: Reject sections with no files, and display no files found instead if no
//sections have any files
class SectionList extends Component {
  state = {}
  render() {
    const { sections, mobile, openSection } = this.props
    const visibleSections = Object.entries(sections).map(([id, section]) => (
      <List.Item
        key={`section-${id}`}
        style={{ display: 'flex', paddingTop: '0.4em', paddingBottom: '0.4em' }}
        onMouseEnter={() => this.setState({ [id]: true })}
        onMouseLeave={() => this.setState({ [id]: false })}
      >
        <List.Icon name="folder" />
        <List.Content>
          {section.name}
          {/* <Button icon="info circle" size="mini" floated="right" basic /> */}
          <FileList section={id} mobile={mobile} />
        </List.Content>
        <Button
          floated="right"
          icon
          size="mini"
          onClick={() => openSection(id)}
          style={
            this.state[id] || mobile
              ? { height: '2.75em', marginRight: '0.5em' }
              : {
                  visibility: 'hidden',
                }
          }
        >
          <Icon name="info circle" />
        </Button>
      </List.Item>
    ))
    return (
      <List
        style={{ flexGrow: 1, overflowY: 'auto', height: '96%' }}
        relaxed
        divided
      >
        {visibleSections}
      </List>
    )
  }
}

SectionList.propTypes = {
  sections: PropTypes.object.isRequired,
  mobile: PropTypes.bool.isRequired,
  openSection: PropTypes.func.isRequired,
}

export default SectionList
