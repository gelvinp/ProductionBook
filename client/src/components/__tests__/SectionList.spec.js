import React from 'react'
import SectionList from '../SectionList.js'
import { shallow } from 'enzyme'

describe('SectionList', () => {
  it('Renders provided sections', () => {
    const name = 'Sample Section'
    const sections = {
      1: {
        name,
        files: {},
      },
    }
    const openButton = jest.fn()
    const wrapper = shallow(
      <SectionList
        sections={sections}
        mobile={false}
        openSection={openButton}
      />
    )
    expect(wrapper.find('ListItem').key()).toBe('section-1')
    expect(wrapper.find('ListIcon').props().name).toBe('folder')
    expect(
      wrapper
        .find('ListContent')
        .childAt(0)
        .text()
    ).toBe(name)
    expect(wrapper).toMatchSnapshot()
    wrapper.find('Button').simulate('click')
    expect(openButton.mock.calls.length).toBe(1)
  })

  it('Shows a button when moused over', () => {
    const name = 'Sample Section'
    const sections = {
      1: {
        name,
        files: {},
      },
    }
    const wrapper = shallow(
      <SectionList sections={sections} mobile={false} openSection={jest.fn()} />
    )
    expect(wrapper.find('Button').props().style.visibility).toEqual('hidden')
    wrapper.find('ListItem').simulate('mouseEnter')
    expect(wrapper.find('Button').props().style).not.toContain('visibility')
    wrapper.find('ListItem').simulate('mouseLeave')
    expect(wrapper.find('Button').props().style.visibility).toEqual('hidden')
  })
})
