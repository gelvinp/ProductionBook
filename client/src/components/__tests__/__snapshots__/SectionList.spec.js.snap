// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SectionList Renders provided sections 1`] = `
<List
  divided={true}
  relaxed={true}
  style={
    Object {
      "flexGrow": 1,
      "height": "96%",
      "overflowY": "auto",
    }
  }
>
  <ListItem
    key="section-1"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    style={
      Object {
        "display": "flex",
        "paddingBottom": "0.4em",
        "paddingTop": "0.4em",
      }
    }
  >
    <ListIcon
      name="folder"
    />
    <ListContent>
      Sample Section
      <Connect(FileList)
        mobile={false}
        section="1"
      />
    </ListContent>
    <Button
      as="button"
      floated="right"
      icon={true}
      onClick={[Function]}
      size="mini"
      style={
        Object {
          "visibility": "hidden",
        }
      }
    >
      <Icon
        as="i"
        name="info circle"
      />
    </Button>
  </ListItem>
</List>
`;
